NativeCharList/Assets/NativeStringTester.cs

```diff
@@ -14,31 +14,16 @@ public class NativeStringTester : MonoBehaviour
 
         private void Start()
         {
-            NativeList<char> test = new NativeList<char>(0, Allocator.Persistent);
-            Debug.Log("Capacity: " + test.Capacity + "\nLength: " + test.Length);
-            test.Add('a');
-            Debug.Log("Capacity: " + test.Capacity + "\nLength: " + test.Length);
-            test.Add('b');
-            Debug.Log("Capacity: " + test.Capacity + "\nLength: " + test.Length);
-            test.Add('c');
-            Debug.Log("Capacity: " + test.Capacity + "\nLength: " + test.Length);
-            test.Dispose();
+            NativeString t1 = (NativeString)"Hello";
```

Comment:
Not sure how I feel about casting this directly anymore.  I foresee issues in the future where we might change t1 to be something else without disposing, so for now I've actually removed the explicit cast in the derivative branch where I debugged indexing... I don't know maybe the explicit cast is enough...  I do like the shorter initialization, I just wish it was clear it was making an entirely new NativeString that will need to be disposed separately.